test/cli/exception-loop/test.out

```diff
@@ -21,18 +18,10 @@ test/cli/exception-loop/test.rb:25: Changing the type of a variable is not permi
   Original type from:
     test/cli/exception-loop/test.rb:17:
     17 |  rescue A => e
-          ^^^^^^
-    test/cli/exception-loop/test.rb:17:
-    17 |  rescue A => e
-                 ^
+                      ^
     test/cli/exception-loop/test.rb:14: Possibly uninitialized (`NilClass`) in:
     14 |def example1
         ^^^^^^^^^^^^
-  Note:
-    The exception variables of two `rescue` blocks conflict with each other because
-    the second is inside a loop. Either use `T.let` to initialize the exception
-    variable before the first `rescue`, or pick unique variable names.
```

Comment:
it looks like we've regressed this test. Any chance we can special case it?

Reply:
I was able to get this behavior back by adding a special case for when the RHS is `<exceptionValue>`.